pkg/lintcontext/parse_yaml.go

```diff
@@ -322,3 +326,58 @@ func normalizeDirectoryPaths(renderedFiles map[string]string) map[string]string
 	}
 	return normalizedFiles
 }
+
+func (l *lintContextImpl) loadObjectsFromKustomize(dir string) {
+	// Create a kustomize engine with source annotations enabled
+	e, err := engine.Kustomize(kustomize.Source{
+		Path: dir,
+	},
+		kustomize.WithSourceAnnotations(true),
+		kustomize.WithWarningHandler(func(warnings []string) error {
+			// Ignore warnings, similar to how we suppress Helm logs with nopWriter
+			return nil
+		}),
+	)
```

Comment:
Readability nit (the indentation looks strange in the review tab)
```suggestion
	kustomizeSource := kustomize.Source{Path: dir}
	ignoreWarnings := func(warnings []string) error {
		// Ignore warnings, similar to how we suppress Helm logs with nopWriter
		return nil
	}
	e, err := engine.Kustomize(
		kustomizeSource,
		kustomize.WithSourceAnnotations(true),
		kustomize.WithWarningHandler(ignoreWarnings),
	)
```